electrumsv/wallet.py

```diff
@@ -550,21 +550,29 @@ def get_public_keys(self, address: Address):
         return self.get_pubkeys(*sequence)
 
     def add_verified_tx(self, tx_hash, height, timestamp, position, proof_position, proof_branch):
-        entry = self._datastore.tx.get_entry(tx_hash, TxFlags.StateCleared)
+        entry = self._datastore.tx.get_entry(tx_hash, TxFlags.StateCleared)  # HasHeight
+
         # Ensure we are not verifying transactions multiple times.
         if entry is None:
-            entry = self._datastore.tx.get_entry(tx_hash)
-            self.logger.debug("Attempting to clear unsettled tx %s %r",
-                tx_hash, entry)
-            return
+            # We have proof now so regardless what TxState is, we can 'upgrade' it to StateSettled.
+            # This rests on the commitment that any of the following four tx States *will*
+            # Have tx bytedata i.e. "HasByteData" flag is set.
+            possible_states = (TxFlags.StateSigned | TxFlags.StateDispatched |
```

Comment:
You already know it's not cleared here, so can exclude that from the possible state flags (which acts as an implicit mask). But the question is, if entry is `None`, shouldn't you do something, like the old "clear unsettled" message?